packages/hooks/src/useCreateOffer.ts

```diff
@@ -79,10 +62,51 @@ export default function useCreateOffer(
   },
 ] {
   const { sdk } = useContext(LiteflowContext)
-  const [transactionHash, setTransactionHash] = useState<string>()
   const [activeStep, setActiveProcess] = useState<CreateOfferStep>(
     CreateOfferStep.INITIAL,
   )
+  const [
+    approveCurrency,
+    {
+      activeStep: approveCurrencyActiveStep,
+      transactionHash: approveCurrencyTransactionHash,
+    },
+  ] = useApproveCurrency(signer)
+  const [
+    approveCollection,
+    {
+      activeStep: approveCollectionActiveStep,
+      transactionHash: approveCollectionTransactionHash,
+    },
+  ] = useApproveCollection(signer)
+
+  // sync approve currency active step
+  useEffect(() => {
+    switch (approveCurrencyActiveStep) {
+      case ApproveCurrencyStep.PENDING: {
+        setActiveProcess(CreateOfferStep.APPROVAL_PENDING)
+        break
+      }
+      case ApproveCurrencyStep.SIGNATURE: {
+        setActiveProcess(CreateOfferStep.APPROVAL_SIGNATURE)
+        break
+      }
+    }
+  }, [approveCurrencyActiveStep])
+
+  // sync approve collection active step
+  useEffect(() => {
+    switch (approveCollectionActiveStep) {
+      case ApproveCollectionStep.PENDING: {
+        setActiveProcess(CreateOfferStep.APPROVAL_PENDING)
+        break
+      }
+      case ApproveCollectionStep.SIGNATURE: {
+        setActiveProcess(CreateOfferStep.APPROVAL_SIGNATURE)
+        break
+      }
+    }
+  }, [approveCollectionActiveStep])
```

Comment:
> Similar comment for the rest.

How does this apply to the `useAcceptOffer` hook? It's has more steps:
```
  TRANSACTION_SIGNATURE,
  TRANSACTION_PENDING,
  OWNERSHIP,
```
A new state with a boolean is not going to work.

Maybe the 2 useEffect could be merged into one with ifs rather than switch case.

> Also, when using a useEffect we need to make sure to have the cleanup function when dealing with states
This is on purpose. The useEffect should not change the step when it's unmounted otherwise it's going to reset the activeStep to initial, which is not what we want if we are on the `SIGNATURE` step.

Reply:
I created issue #130 about that, this is not blocking this PR, as everything works well. We could have local state and have a general memo "converting" all the states but let's do that in another step as it is not required